benchmarks/continous/db/tool/orm/migrations/2024-06-12-120242_refine_ft_transfers/down.sql

```diff
@@ -0,0 +1,23 @@
+-- Assumes regularly values handled here cannot be negative, hence -1 flags out
+-- of range values.
+create function convert_bigint_to_int(x bigint)
+returns integer as $$
+begin
+    if x between -2147483648 and 2147483647 then
```

Comment:
Given that we store `size_state_bytes` in i64, shouldn't these limits be for i64 (-2**63, 2**63-1) instead of the ones for i32 used here?

Reply:
I guess I'm generally confused about the purpose of this function. Is it used always or only during migration?

Reply:
We want to convert a `bigint` to `integer`, which is fine as long as `x` fits into `integer`. Accordingly the function does:

```
if x::bigint is in range of integer
  return x::integer
else x::bigint is out of integer range
  return -1 to signal the conversion failed
```

Once this migration succeeded, I don't expect us to ever run this `down.sql`. So I think there is no need to handle it in a more sophisticated way.

Reply:
> Is it used always or only during migration?

Only if this migration is roled back. I think it's common to use `migrations/migration_x/down.sql` to roll back a migration. At least `diesel` uses `down.sql` files for that purpose.

Reply:
[e7bd34f](https://github.com/near/nearcore/pull/11563/commits/e7bd34f871ef7adaaf47dcc3e4161f5abeb16ea5) extends the README's section on migrations. Hope that helps to clarify the purpose of these files.